### PR TITLE
LTP: Fix for fcntl24,fcntl25,fcntl26,fcntl32

### PR DESCRIPTION
**Pull request by ashokchandran**
Issue: Test was failing as fcntl was not supported on RAMFS. 
Solution: Use root filesystem to test

### CODE REVIEW

File: tests/ltp/patches/fcntl24.patch [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-
+     /***************************************************************
+      * check looping state if -c option given
+      ***************************************************************/
+@@ -185,13 +181,12 @@ void setup(void)
+ {
+ 
+ 	tst_sig(NOFORK, DEF_HANDLER, cleanup);
++	rmdir(MNTPOINT);
++	SAFE_MKDIR(cleanup,MNTPOINT, DIR_MODE);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl24.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl24.patch [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
++	SAFE_MOUNT(cleanup,device, MNTPOINT, fs_type, 0, NULL);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl24.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl24.patch [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ 			 "open(%s, O_RDWR|O_CREAT,0777) Failed, errno=%d : %s",
+ 			 fname, errno, strerror(errno));
+@@ -210,7 +205,8 @@ void cleanup(void)
+ 		tst_resm(TWARN, "close(%s) Failed, errno=%d : %s", fname, errno,
+ 			 strerror(errno));
+ 	}
+-
+-	tst_rmdir();
++	remove(FILENAME);
++	SAFE_UMOUNT(NULL,MNTPOINT);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl24.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl24.patch [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
++	SAFE_RMDIR(NULL,MNTPOINT);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl24.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl25.patch [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+     /***************************************************************
+      * check looping state if -c option given
+      ***************************************************************/
+@@ -186,13 +183,11 @@ void setup(void)
+ {
+ 
+ 	tst_sig(NOFORK, DEF_HANDLER, cleanup);
++	rmdir(MNTPOINT);
++	SAFE_MKDIR(cleanup,MNTPOINT, DIR_MODE);
++	SAFE_MOUNT(cleanup,device, MNTPOINT, fs_type, 0, NULL);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl25.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl25.patch [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ 			 fname, errno, strerror(errno));
+@@ -211,7 +206,8 @@ void cleanup(void)
+ 		tst_resm(TWARN, "close(%s) Failed, errno=%d : %s", fname, errno,
+ 			 strerror(errno));
+ 	}
+-
+-	tst_rmdir();
++	remove(FILENAME);
++	SAFE_UMOUNT(NULL,MNTPOINT);
++	SAFE_RMDIR(NULL,MNTPOINT);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl25.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl26.patch [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+     /***************************************************************
+      * check looping state if -c option given
+      ***************************************************************/
+@@ -186,13 +183,11 @@ void setup(void)
+ {
+ 
+ 	tst_sig(NOFORK, DEF_HANDLER, cleanup);
++	rmdir(MNTPOINT);
++	SAFE_MKDIR(cleanup,MNTPOINT, DIR_MODE);
++	SAFE_MOUNT(cleanup,device, MNTPOINT, fs_type, 0, NULL);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl26.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl26.patch [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+ 			 "open(%s, O_WRONLY|O_CREAT,0777) Failed, errno=%d : %s",
+ 			 fname, errno, strerror(errno));
+@@ -212,6 +207,9 @@ void cleanup(void)
+ 			 strerror(errno));
+ 	}
+ 
+-	tst_rmdir();
++	remove(FILENAME);
++	SAFE_UMOUNT(NULL,MNTPOINT);
++	SAFE_RMDIR(NULL,MNTPOINT);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl26.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl32.patch [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+-		tst_brkm(TCONF, cleanup,
+-			 "Cannot do fcntl(F_SETLEASE, F_WRLCK) "
+-			 "on %s filesystem",
+-			 tst_fs_type_name(type));
+-	default:
+-		break;
+-	}
++	rmdir(MNTPOINT);
++	SAFE_MKDIR(cleanup,MNTPOINT, DIR_MODE);
++	SAFE_MOUNT(cleanup,device, MNTPOINT, fs_type, 0, NULL);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl32.patch]: Space after comma in function calls

File: tests/ltp/patches/fcntl32.patch [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+ 
+ 	TEST(fcntl(fd1, F_SETLEASE, F_WRLCK));
+ 
+@@ -132,6 +125,8 @@ static void cleanup(void)
+ 
+ 	if (fd2 > 0 && close(fd2))
+ 		tst_resm(TWARN | TERRNO, "Failed to close file");
++	remove(FILENAME);
++	SAFE_UMOUNT(NULL,MNTPOINT);
++	SAFE_RMDIR(NULL,MNTPOINT);
```
Comment (hukoyu) [inline tests/ltp/patches/fcntl32.patch]: Space after comma in function calls

**Review by hukoyu — CHANGES_REQUESTED** [review]
@ashokchandran please put space after comma in function calls. Everyting else looks good.

**Review by shaikshavali1 — COMMENTED** [review]
@ashokchandran, I don't have any additional comments apart from Huseyin's comments.

File: tests/ltp/patches/fcntl24.patch (hunk 1 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl24.patch]: done

File: tests/ltp/patches/fcntl24.patch (hunk 2 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl24.patch]: done

File: tests/ltp/patches/fcntl24.patch (hunk 4 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl24.patch]: done

File: tests/ltp/patches/fcntl25.patch (hunk 5 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl25.patch]: done

File: tests/ltp/patches/fcntl25.patch (hunk 6 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl25.patch]: done

File: tests/ltp/patches/fcntl26.patch (hunk 7 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl26.patch]: done

File: tests/ltp/patches/fcntl26.patch (hunk 8 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl26.patch]: done

File: tests/ltp/patches/fcntl24.patch (hunk 3 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl24.patch]: done

File: tests/ltp/patches/fcntl32.patch (hunk 9 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl32.patch]: done

File: tests/ltp/patches/fcntl32.patch (hunk 10 above)
Comment (ashokchandran) [reply tests/ltp/patches/fcntl32.patch]: done

**Review by hukoyu — APPROVED** [review]
LGTM

**Pull request merged**